hazelcast/vector.py

```diff
@@ -68,20 +102,36 @@ class Metric(enum.IntEnum):
 
 @dataclasses.dataclass
 class IndexConfig:
+    """The IndexConfig contains configuration for a vector index.
+
+    Args:
+        name: Name of the vector index.
+        metric: The metric to be used for the index configuration.
+        dimension: The dimension of vectors to be used in the vector index.
+            All vectors that use this index configuration must have
+            the same dimension.
+        max_degree: The maximum number of connections allowed per node.
+        ef_construction: The size of the dynamic list for search.
+        use_deduplication: Enable deduplication in the index.
+    """
 
     name: str
     metric: Metric
     dimension: int
-    max_degree: int = 200
-    ef_construction: int = 300
-    use_deduplication: bool = False
+    max_degree: int = 16
+    ef_construction: int = 100
+    use_deduplication: bool = True
 
 
 VectorIndexConfig = IndexConfig
 
 
 @dataclasses.dataclass
 class VectorSearchOptions:
+    """VectorSearchOptions contains search configuration options.
```

Comment:
I think we can make this an inner class in VectorCollection. Not sure if it appears in the python API reference if we do that. It does not have to appear, so we can make this internal if possible

Reply:
Why make it an inner class of VectorCollection?